eZ/Publish/Core/settings/search_engines/legacy/criterion_handlers_common.yml

```diff
@@ -65,6 +65,8 @@ services:
     ezpublish.search.legacy.gateway.criterion_handler.common.content_type_identifier:
         parent: ezpublish.search.legacy.gateway.criterion_handler.base
         class: %ezpublish.search.legacy.gateway.criterion_handler.common.content_type_identifier.class%
+        arguments:
+            - @ezpublish.spi.persistence.content_type_handler
```

Comment:
You're missing `@ezpublish.api.storage_engine.legacy.dbhandler` as 1st argument.